lib/services/API/country_educational_institute_api.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:mobile_app/constants.dart';
+import 'package:mobile_app/models/failure_model.dart';
+import 'package:mobile_app/utils/api_utils.dart';
+
+abstract class CountryInstituteAPI {
+  Future<List<String>> getSuggestions(String query, String url);
+}
+
+class HttpCountryInstituteAPI implements CountryInstituteAPI {
+  bool _countryFlag = true;
+  bool _instituteFlag = true;
+  List<dynamic> data;
+
+  @override
+  Future<List<String>> getSuggestions(String query, String value) async {
+    var url = (value == Constants.COUNTRY)
+        ? 'https://restcountries.eu/rest/v2/all'
+        : 'http://universities.hipolabs.com/search?';
+
+    var headers = {'Content-Type': 'application/json'};
+    try {
+      if ((_countryFlag && value == Constants.COUNTRY) ||
+          (_instituteFlag && value == Constants.EDUCATIONAL_INSTITUTE)) {
```

Comment:
This if condition statement is redundant.

Reply:
Again , this condition is to ensure that only one call is made to API.